packages/devextreme/js/__internal/grids/grid_core/keyboard_navigation/m_keyboard_navigation.ts

```diff
@@ -2052,7 +2052,13 @@ export class KeyboardNavigationController extends modules.ViewController {
       if (!keyPressEvent.isDefaultPrevented()) {
         const timeout = browser.mozilla ? 25 : 0; // T882996
         setTimeout(() => {
-          $input.val(editorValue);
+          let inputValue = editorValue;
+          // @ts-expect-error
+          if (editorValue === '-' && $input.val() === '-0') { // T1203026
+            inputValue = '-0';
+          }
+
+          $input.val(inputValue);
```

Comment:
Let's move this "crutch" code block to a separate method.
It makes it more isolated and easy to locate and delete in the future.

Also, let's get rid of the ```@ts-expect-error``` and add more detailed ```// NOTE: ...``` comment about things that happens here :)

E.g.
Move to a separate method:
```typescript
  // NOTE: See the T1203026 ticket.
  // Please add more description here :)
  _getKeyPressInputValue(
    $input: dxElementWrapper,
    editorValue: any,
  ): any {
    const inputCurrentValue: any = $input.val();

    return editorValue === '-' && inputCurrentValue === '-0'
      ? '-0'
      : editorValue;
  }
```

And usage of this method:
```suggestion
          const inputValue = this._getKeyPressInputValue($input, editorValue);
          $input.val(inputValue);
```